lambdas/package/index.py

```diff
@@ -0,0 +1,186 @@
+"""
+Provide a virtual-file-system view of a package's logical keys.
+"""
+
+import io
+import json
+
+import boto3
+import pandas as pd
+
+from t4_lambda_shared.decorator import api, validate
+from t4_lambda_shared.utils import get_default_origins, make_json_response
+
+S3_DOMAIN_SUFFIX = '.amazonaws.com'
+
+SCHEMA = {
+    'type': 'object',
+    'properties': {
+        'bucket': {
+            'type': 'string'
+        },
+        'manifest': {
+            'type': 'string'
+        },
+        'access_key': {
+            'type': 'string'
+        },
+        'secret_key': {
+            'type': 'string'
+        },
+        'session_token': {
+            'type': 'string'
+        },
+        'prefix': {
+            'type': 'string'
+        },
+        'logical_key': {
+            'type': 'string'
+        }
+    },
+    'required': ['bucket', 'manifest', 'access_key', 'secret_key'],
+    'additionalProperties': False
+}
+
+
+class IncompleteResultException(Exception):
+    """
+    Exception indicating an incomplete response
+    (e.g., from S3 Select)
+    """
+
+
+def buffer_s3response(s3response):
+    """
+    Read a streaming response from s3 select into a
+    StringIO buffer
+    """
+    response = io.StringIO()
+    end_event_received = False
+    stats = None
+    for event in s3response['Payload']:
```

Comment:
I would use this logic in the indexer lambda (also relies on S3 select, so you could put this in shared uitls.py? `get_s3_select_data` or something.

Reply:
Done.